substreams/ethereum-nabla/src/modules/store_components.rs

```diff
@@ -0,0 +1,35 @@
+use crate::modules::map_components::ComponentType;
+use crate::pb::nabla::v1::{component::Kind, Component, Portal, Router, SwapPool};
+use substreams::store::{StoreNew, StoreSetIfNotExists, StoreSetIfNotExistsProto};
+use tycho_substreams::prelude::{BlockTransactionProtocolComponents, ProtocolComponent};
+
+impl From<ProtocolComponent> for Component {
+    fn from(pc: ProtocolComponent) -> Self {
+        let protocol_type = pc
+            .protocol_type
+            .expect("ProtocolType not set")
+            .name;
+        let kind = match protocol_type.into() {
+            ComponentType::Portal => Kind::Portal(Portal { address: pc.id.into() }),
+            ComponentType::Router => Kind::Router(Router { address: pc.id.into() }),
+            ComponentType::SwapPool => Kind::SwapPool(SwapPool { address: pc.id.into() }),
+        };
+        Component { kind: Some(kind) }
+    }
+}
```

Comment:
The `protocol_type` cannot be `None`, however it is returned as an `Option` due to how `.proto` file is translated into rust code for `OneOf`, hence `.expect()` is fine here.